src/Utilities/functions.test.js

```diff
@@ -0,0 +1,158 @@
+    });
+
+    it('should return visible schema from store', () => {
+        const schema = { fields: []};
+        const section = getSection('first', { isVisible: true, title: 'Some title' }, { loaded: true, schema });
+        expect(section).toMatchObject({
+            label: 'Some title',
+            name: 'first',
+            fields: schema
+        });
+    });
+
+    it('should return loader', () => {
+        const section = getSection();
+        const wrapper = mount(section.label);
+        expect(toJson(wrapper)).toMatchSnapshot();
+        expect(section).toMatchObject({ fields: loaderField });
+    });
+
+    it('should call success function', (done) => {
+        const success = jest.fn();
+        const section = getSection('first', { isVisible: false }, {}, success);
+        expect(section).toMatchObject({
+            name: 'first',
+            fields: loaderField
+        });
+        setTimeout(() => {
+            expect(success).toHaveBeenCalled();
+            done();
+        });
+    });
+});
+
+describe('calculatePermissions', () => {
+    it('should check visibility of one function', async () => {
+        const isVisible = await calculatePermissions({
+            method: 'something'
+        });
+        expect(isVisible).toBe(false);
+    });
+
+    it('should check visibility of array of functions', async () => {
+        const isVisible = await calculatePermissions([{
+            method: 'something'
+        }, {
+            method: 'something',
+            args: [ true ]
+        }]);
+        expect(isVisible).toBe(false);
+    });
+
+    it('should calculate visible for one function', async () => {
+        const isVisible = await calculatePermissions([{
+            method: 'something',
+            args: [ true ]
+        }]);
+        expect(isVisible).toBe(true);
+    });
+});
+
+describe('calculateEmailConfig', () => {
+    it('should not throw error', async () => {
+        const result = await calculateEmailConfig();
+        expect(result).toMatchObject({});
+    });
+
+    it('should calculate schema with permissions - false', async () => {
+        mock.onGet('/api*').reply(200, {});
+        const result = calculateEmailConfig({
+            'email-preference': {
+                test: {
+                    permissions: { method: 'something' }
+                }
+            }
+        });
+        expect(await result.test.isVisible).toBe(false);
+    });
+
+    it('should calculate schema with permissions - true', async () => {
+        mock.onGet('/api/test/v1/user-config/email-preference').reply(200, {});
+        const result = calculateEmailConfig({
+            'email-preference': {
+                test: {
+                    permissions: { method: 'something', args: [ true ]}
+                }
+            }
+        });
+        expect(await result.test.isVisible).toBe(true);
+    });
+
+    it('should request localFile', (done) => {
```

Comment:
why not to use async / await here too?

Reply:
Because the dispatch function is ran in paralel and is not waited for. Hence the usage of `setImmediate` where we have to wait for the async function to fullfill.